services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/MultipartManagerV2Impl.java

```diff
@@ -147,7 +145,18 @@ <T extends MultipartRequest> MultipartUploadStatus startOrResumeMultipartRequest
 		
 		return status.getMultipartUploadStatus();
 	}
-	
+
+	private boolean isNewUploadRequestRequired(boolean forceRestart, UserInfo user, String requestMD5Hex, CompositeMultipartUploadStatus status) {

```

Comment:
this method seems like a boolean question but it has an unexpected side-effect (it automatically re-starts the upload).

With a function that only returns a boolean (no side-effects):
then
if(forceRestart || fileHandleExistsFileNoLongerInCloudeStorage()){
        resetMultiPart()
}